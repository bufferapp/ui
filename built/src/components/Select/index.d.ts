export { default } from './Select';
//# sourceMappingURL=index.d.ts.map