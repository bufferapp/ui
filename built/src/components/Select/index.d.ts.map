{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/Select/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,UAAU,CAAA"}