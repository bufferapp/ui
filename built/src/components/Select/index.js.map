{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/Select/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,UAAU,CAAA","sourcesContent":["export { default } from './Select'\n"]}