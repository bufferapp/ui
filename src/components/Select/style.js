/* eslint-disable no-nested-ternary */
import styled from 'styled-components';
import {
  grayShadow,
  grayLighter,
  white,
  gray,
  grayLight,
  grayDarker,
} from '../style/colors';
import { fontFamily } from '../style/fonts';

export const Wrapper = styled.div`
  outline-style: none;
  :focus {
    outline-style: 0;
  }
  width: ${(props) => (props.isSplit || props.fullWidth ? '100%' : 'auto')};
  height: ${(props) => (props.isSplit ? '100%' : 'auto')};
  max-height: ${(props) => (props.isSplit ? '100%' : '')};
  position: ${(props) => (props.isSplit ? 'initial' : 'relative')};
  display: inline-block;
`;

export const SelectStyled = styled.div`
  right: ${props => (props.hasIconOnly ? '6px' : '0')};
  position: absolute;
  z-index: 1000;
  border: 1px solid ${gray};
  box-sizing: border-box;
  box-shadow: ${grayShadow};
  border-radius: 4px;
  display: ${props => (props.isOpen ? 'initial' : 'none')};
  min-width: 200px;
  max-width: ${props => (props.fullWidth ? '' : '200px')};
  width: ${props => (props.fullWidth ? '100%' : '200px')};
  background-color: ${white};
  bottom: ${props => (props.yPosition === 'top' ? '100%' : 'initial')};
  top: ${props => (props.yPosition === 'bottom' ? '100%' : 'initial')};
  margin-bottom: ${props => (props.yPosition === 'top' ? '8px' : '0')};
  margin-top: ${props =>
    props.marginTop
      ? props.marginTop
      : props.yPosition === 'bottom' || props.xPosition === 'right'
      ? '8px'
      : '0'};
  transform: translateX(
    ${props => (props.horizontalOffset ? props.horizontalOffset : '0')}
  );
  left: ${props => (props.xPosition === 'left' ? 0 : '')};
`;

export const SelectItems = styled.ul`
  max-height: 195px;
  overflow-y: auto;
  position: relative;
  z-index: -1;
  background: ${white};
  border-radius: 4px;
  list-style: none;
  margin: 8px 0px;
  padding: 0;
`;

export const SelectItemDivider = styled.li`
  background-color: ${grayLighter};
  height: 1px;
  margin-bottom: 8px;
  margin-top: 8px;
  padding: 0;
  pointer-events: none;
  width: 100%;
`;

export const SelectItemDividerTitle = styled.span`
  position: relative;
  top: -11px;
  margin-left: 8px;
  background: white;
  padding: 0 8px;
  font-size: 12px;
  font-family: ${fontFamily};
  color: ${gray};
`;

export const SearchBarWrapper = styled.div`
  display: flex;
  border-bottom: 1px solid #ecf0f3;
  align-items: center;
  padding-left: 16px;
  color: ${gray};
`;

export const NoMatchesContainer = styled.p`
  width: 116px;
  height: 17px;
  margin: 0 auto;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  line-height: normal;
  font-size: 14px;
  letter-spacing: -0.233333px;
  color: ${gray};
`;

export const CustomItemContainer = styled.li`
  min-height: 32px;
  align-items: center;
  color: ${grayDarker};
  cursor: pointer;
  display: flex;
  font-size: 14px;
  font-weight: 500;
  overflow: hidden;
  text-overflow: ellipsis;
  user-select: none;
  white-space: nowrap;
  margin-left: 8px;
  margin-right: 8px;
  border-radius: 4px;
  padding-left: 8px;
  padding-right: 8px;
  &:hover {
    background-color: ${grayLight};
  }
  background-color: ${props =>
    props.isCustomItemFocused ? `${grayLight}` : ''};
`;
