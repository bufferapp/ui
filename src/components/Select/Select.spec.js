import snap from 'jest-auto-snapshots'
import Select from './Select'

snap(Select, './Select.jsx')
