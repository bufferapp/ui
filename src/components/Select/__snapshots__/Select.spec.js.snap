// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Select Matches snapshot when array prop "hotKeys" has one item: "[object Object]" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when array prop "hotKeys" is an empty array: "[]" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when array prop "items" has one item: "[object Object]" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "capitalizeItemLabel" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={false}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "clearSearchOnBlur" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "disabled" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip="jest-auto-snapshots String Fixture"
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={false}
    onClick={[Function]}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="white"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "fullWidth" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={false}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={false}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "hasCustomAction" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "hasIconOnly" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={false}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "hideSearch" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={false}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "isInputSearch" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "isOpen" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={false}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "isSplit" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={false}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "multiSelect" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={false}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "selectPopupVisible" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when boolean prop "shortcutsEnabled" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when passed all props 1`] = `
<ForwardRef(styled.div)
  aria-expanded={true}
  aria-haspopup="true"
  data-tip=""
  fullWidth={true}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <ForwardRef(styled.div)
    disabled={true}
    size="small"
    type="primary"
  >
    <ChevronDownIcon
      color="grayDark"
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Select Matches snapshot when passed only required props 1`] = `
<ForwardRef(styled.div)
  aria-expanded={false}
  aria-haspopup="true"
  isSplit={false}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <Button
    disabled={false}
    fullWidth={false}
    hasIconOnly={false}
    hideSearch={true}
    iconEnd={false}
    isSelect={true}
    items={
      Array [
        Object {
          "hideOnSearch": true,
          "id": "jest-auto-snapshots String Fixture",
          "title": "jest-auto-snapshots String Fixture",
        },
      ]
    }
    label=""
    loading={false}
    onClick={[Function]}
    selectPosition="bottom"
    size="medium"
    type="secondary"
  />
  <ForwardRef(styled.div)
    hasIconOnly={false}
    isOpen={false}
    xPosition="left"
    yPosition="bottom"
  >
    <ForwardRef(styled.ul)>
      <SelectItemTooltipWrapper
        capitalizeItemLabel={true}
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={false}
        hovered={false}
        item={
          Object {
            "hideOnSearch": true,
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={false}
        onClick={[Function]}
        onItemClick={[Function]}
      />
    </ForwardRef(styled.ul)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;
